packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1817,6 +1817,14 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     _updateHandleLayersAndOwners();
   }
 
+  Rect _getBoundingBox(Selectable selectable) {
```

Comment:
If we merge before sorting, wouldn't we run into corner cases like before?

Reply:
I realized this is probably come from my previous suggestions. I am not sure what the right answer is.

Reply:
I think merging before sorting is okay, the corner cases we were running into were mostly due to some invalid logic I added into the logic that compares selectables, so the selectables were out of order. See https://github.com/flutter/flutter/pull/136920#discussion_r1419482870 .

Another issue was that even when the selectables are in order we would still hit selectables that did not actually contain the tapped global position. boundingBoxes solves this by giving us the precise rects to check against versus an all encompassing rect that we had before.

Take `Text before widgetspan [SelectableWidgetSpan] Text after widgetspan`.

We want to select something in `Text after widgetspan`.
 
SelectableA: `Text before widgetspan`
SelectableB: `[SelectableWidgetSpan]`
SelectableC: `Text after widgetspan`
Order: [SelectableA, SelectableB, SelectableC]

SelectableA's rect would encompass SelectableB and SelectableC. So a tap on either of those Selectables will always hit SelectableA first.

In the example above we would still hit SelectableA, and SelectableB before hitting SelectableC. SelectableA would return SelectionResult.next as expected, but the following SelectableB would not contain the global position at all so the iteration would stop there and never reach SelectableC. 

By iterating through the boundingBoxes we will now only hit SelectableC.

Reply:
> SelectableA's rect would encompass SelectableB and SelectableC. So a tap on either of those Selectables will always hit SelectableA first.

How so? is the `Text before widgetspan` got broken into two lines?

Reply:
Oh yeah sorry forgot that detail. The example only makes sense if `Text before widgetspan` is on two lines.

We wouldn't run into this issue in the case where all Selectable fragments are on the same line because they do not overlap.